core/classes/BugFilterQuery.class.php

```diff
@@ -1518,11 +1518,13 @@ protected function helper_sort_column_alias( $p_prop ) {
 			case 'category_id':
 				# This join will be reduced as unique, if category search is active
 				$this->add_join( 'LEFT JOIN {category} ON {bug}.category_id = {category}.id' );
+				$this->add_select( '{category}.name' );
 				return '{category}.name';
```

Comment:
Why do you need to add `{category}.name` in the select and also return it?

Reply:
> Why do you need to add {category}.name in the select and also return it?

This function returns the alias that must be used in the sort clause. Note that this function does not build the order by clauses, however, it does manages the additional join/select stuff needed to be able to use that sorting alias.